packages/react-jss/src/withStyles.js

```diff
@@ -65,7 +65,8 @@ export default function withStyles<Theme: {}, S: Styles<Theme>>(
     InnerComponent: ComponentType<Props> = NoRenderer
   ): ComponentType<Props> => {
     const displayName = getDisplayName(InnerComponent)
-    const defaultClassNamePrefix = process.env.NODE_ENV === 'production' ? '' : `${displayName}-`
+    const defaultClassNamePrefix =
+      process.env.NODE_ENV === 'production' ? '' : `${displayName.replace(/\s/, '-')}-`
```

Comment:
- we should match all spaces, not just one, global flag is missing
- lets add a test